airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/state/listener/WorkflowStateChangedListener.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.state.listener;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import java.util.Queue;
+import java.util.UUID;
+import lombok.Value;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME,
```

Comment:
are the json annotations because this needs to be serialised in the database?

Reply:
can we add a short doc string explaining this class here?

Reply:
Yes, I need to register the different implementations to jackson because the `ConnectionManagerWorkflowImpl` is using the interface and without that the serialization is not possible.